avm/ptn/authorization/pim-role-assignment/main.bicep

```
@@ -0,0 +1,162 @@
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+      outputs: {
+        telemetry: {
+          type: 'String'
+          value: 'For more information, see https://aka.ms/avm/TelemetryInfo'
+        }
+      }
+    }
+  }
+  location: location
+}
+
+module roleAssignment_mg 'modules/management-group.bicep' = if (empty(subscriptionId) && empty(resourceGroupName)) {
+  name: '${uniqueString(deployment().name, location)}-PimRoleAssignment-MG-Module'
+  scope: managementGroup(managementGroupId)
+  params: {
+    roleDefinitionIdOrName: roleDefinitionIdOrName
+    principalId: principalId
+    managementGroupId: managementGroupId
+    requestType: requestType
+    justification: justification
+    targetRoleEligibilityScheduleId: targetRoleEligibilityScheduleId
+    targetRoleEligibilityScheduleInstanceId: targetRoleEligibilityScheduleInstanceId
+    targetRoleAssignmentScheduleId: targetRoleAssignmentScheduleId
+    targetRoleAssignmentScheduleInstanceId: targetRoleAssignmentScheduleInstanceId
+    ticketInfo: ticketInfo
+    conditionVersion: conditionVersion
+    condition: !empty(condition) ? condition : ''
+    pimRoleAssignmentType: pimRoleAssignmentType
+  }
+}
+
+module roleAssignment_sub 'modules/subscription.bicep' = if (!empty(subscriptionId) && empty(resourceGroupName)) {
+  name: '${uniqueString(deployment().name, location)}-PimRoleAssignment-Sub-Module'
+  scope: subscription(subscriptionId)
+  params: {
+    roleDefinitionIdOrName: roleDefinitionIdOrName
+    principalId: principalId
+    subscriptionId: subscriptionId
+    requestType: requestType
+    justification: justification
+    targetRoleEligibilityScheduleId: targetRoleEligibilityScheduleId
+    targetRoleEligibilityScheduleInstanceId: targetRoleEligibilityScheduleInstanceId
+    ticketInfo: ticketInfo
+    conditionVersion: conditionVersion
+    condition: !empty(condition) ? condition : ''
+    pimRoleAssignmentType: pimRoleAssignmentType
+  }
+}
+
+module roleAssignment_rg 'modules/resource-group.bicep' = if (!empty(resourceGroupName) && !empty(subscriptionId)) {
+  name: '${uniqueString(deployment().name, location)}-PimRoleAssignment-RG-Module'
+  scope: resourceGroup(subscriptionId, resourceGroupName)
+  params: {
+    roleDefinitionIdOrName: roleDefinitionIdOrName
+    principalId: principalId
+    subscriptionId: subscriptionId
+    resourceGroupName: resourceGroupName
+    requestType: requestType
+    justification: justification
+    targetRoleEligibilityScheduleId: targetRoleEligibilityScheduleId
+    targetRoleEligibilityScheduleInstanceId: targetRoleEligibilityScheduleInstanceId
+    ticketInfo: ticketInfo
+    conditionVersion: conditionVersion
+    condition: !empty(condition) ? condition : ''
```

Same as in the other comment
```suggestion
    condition: condition
```

---

Resolved